packages/logging-bunyan/src/index.js

```diff
@@ -131,6 +131,29 @@ LoggingBunyan.prototype.write = function(record) {
     );
   }
 
+  // Stackdriver Logs Viewer picks up the summary line from the 'message' field
+  // of the payload.
+  // https://cloud.google.com/logging/docs/view/logs_viewer_v2#expanding
+  //
+  // Furthermore, for error messages at severity 'error' and higher, Stackdriver
+  // Error Reporting will pick up error messages if the full stack trace is
+  // included in the textPayload or in the `message` property of the
+  // jsonPaylod. See:
+  // https://cloud.google.com/error-reporting/docs/formatting-error-messages
+  //
+  // We only do this if the user hasn't provided a message property themselves.
+  //
+  // TODO(ofrobots): when resource.type is 'global' we need to additionally
+  // provide serviceContext.service as part of the entry for Error Reporting to
+  // automatically pick up the error.
+  if (!record.message) {
+    if (record.err && record.err.stack) {
+      record.message = record.err.stack;
+    } else {
+      record.message = record.msg;
+    }
+  }
+
```

Comment:
Thanks, that's a lot more readable, albeit if we wrote it like this: `record.message = record.message || (record.err && record.err.stack) || record.msg`. However, you raise a good point about `record.msg` and `record.message` being duplicated values in some cases – correctly encoding the logic I want here puts me back to an if-conditional rather than the assignment you suggested:

```
  if (!record.message) {
    if (record.err && record.err.stack) {
      record.message = recored.err.stack;
    } else {
      record.message = record.msg;
      delete record.msg;
    }
  }
```

I'm also going to add a clone of `record` at the top.